core/server/worker/src/main/java/alluxio/worker/block/BlockHeartbeatReport.java

```diff
@@ -55,4 +63,14 @@ public BlockHeartbeatReport(Map<String, List<Long>> addedBlocks, List<Long> remo
   public List<Long> getRemovedBlocks() {
     return Collections.unmodifiableList(mRemovedBlocks);
   }
+
+  /**
+   * Gets the storage paths which were lost
+   * in the heartbeat this report represents.
+   *
+   * @return a map from storage tier alias to lost storage paths
+   */
+  public Map<String, List<String>> getLostStorage() {
+    return Collections.unmodifiableMap(mLostStorage);
```

Comment:
@aaudiber Now only here I use the `unmodifableMap` because the map in the `blockheartbeatReport` is fixed and will never be updated.

Reply:
It's valuable to be able to enforce properties within a single file. Right now the parameters passed to the constructor happen to not be changed externally, but someone could change that in the future without looking at this class. If we copy the parameters in the constructor, this class won't have concurrency issues even if the lists are changed by code outside this class.

Reply:
Make sense, now I move the copy process into the `BlockHeartbeatReport` to confirm that the maps and lists inside `BlockHeartbeatReport` are never modified. Thanks